libs/core/langchain_core/graph_stores/links.py

```diff
@@ -0,0 +1,56 @@
+from dataclasses import dataclass
+from typing import Iterable, Literal, Set, Union
+
+from langchain_core.documents import Document
+
+
+@dataclass(frozen=True)
+class Link:
```

Comment:
Links allow a document to link to (or from) other documents without knowing their IDs. This is important since we may not have all documents presented at the same time. They operate similarly to a hyperlink in HTML – a website may link to another URL without knowing whether it exists or what it will resolve to. The destination is discovered during traversal.

Kind acts as a namespace, and allows different extractors to use the same tag name without creating collisions between extractors. For example “keyword” vs “url”.